packages/s/spdlog/xmake.lua

```diff
@@ -24,25 +24,30 @@ package("spdlog")
 
     if is_plat("windows") then
         add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
+        add_configs("wchar",  {description = "Support wchar api.", default = false, type = "boolean"})
     end
 
     on_load(function (package)
         if not package:config("header_only") then
-            package:add("defines", "SPDLOG_COMPILE_LIB")
+            package:add("defines", "SPDLOG_COMPILED_LIB")
+            if package:version():ge("1.4.0") then
+                package:add("deps", "cmake")
+            end
         end
         if package:config("fmt_external") then
             package:add("defines", "SPDLOG_FMT_EXTERNAL")
+            package:add("deps", "fmt", {configs = {header_only = true}})
         end
-        if not package:config("header_only") then
-            package:add("deps", "cmake")
+        if package:config("noexcept") then
+            package:add("defines", "SPDLOG_NO_EXCEPTIONS")
         end
-        if package:config("fmt_external") then
-            package:add("deps", "fmt", {configs = {header_only = true}})
+        if package:config("wchar") then
+            package:add("defines", "SPDLOG_WCHAR_TO_UTF8_SUPPORT")
         end
     end)
 
     on_install(function (package)
-        if (package:version() and package:version():lt("1.4.0")) or package:config("header_only") then
```

Comment:
package:version() will be nil if it is from git branch, e.g. `add_requires("spdlog master")`

Reply:
Interesting, I was following what other packages do. Does that mean that all packages that can accept a branch name must check whether version is nil?

Reply:
yes, if it provides git url.

and we can also use `package:gitref()` to check it's from git branch.

Reply:
If I understand this correctly, does this mean that even if you specify a commit or tag or branch that also satisfies the condition on the right, being earlier than or equal to 2.5.9, it will fail the `if check`?

Reply:
? if you use `add_requires("xxx branch")` to set branch, xmake cannot get package version. the branch name may be any string.

xmake just can select versions from `add_versions()`